lib/bolt/task.rb

```diff
@@ -33,21 +33,29 @@ def supports_noop
     end
 
     def file_map
-      @file_map ||= files.each_with_object({}) { |file, hsh| hsh[file['name']] = file['path'] }
+      @file_map ||= files.each_with_object({}) { |file, hsh| hsh[file['name']] = file }
     end
     private :file_map
 
+    def file_path(file_name)
+      file_map[file_name]['path']
+    end
+
     # Returns a hash of implementation name, path to executable, input method (if defined),
     # and any additional files (name and path)
     def select_implementation(target, additional_features = [])
+      # TODO: This message isn't correct and this check is probably api only
       raise 'select_implementation only supported with multiple files' if files.nil? || files.empty?
 
       impl = if (impls = metadata['implementations'])
                available_features = target.features + additional_features
                impl = impls.find { |imp| Set.new(imp['requirements']).subset?(available_features) }
                raise "No suitable implementation of #{name} for #{target.name}" unless impl
                impl = impl.dup
-               impl['path'] = file_map[impl['name']]
+               # TODO: I think we may have unintentionally allowed a
```

Comment:
I can't think of a compelling use case for having implementations across multiple modules. Maybe if there were separate modules for doing similar tasks on different platforms, and you couldn't consolidate them into one module? But I don't think that's a common issue, and can't find an example if it on the forge. I also think it could get confusing if you had different modules with the same implementation file names.